pipeline/matching/generators/similar_interests.py

```diff
@@ -1,18 +1,50 @@
+import itertools
 from typing import Iterator
 
 from pipeline.matching.core import MatchGenerator
-from pipeline.types import Match, MatchingInput
+from pipeline.types import Match, MatchingInput, MatchMetadata
 
 GENERATOR_SIMILAR_INTERESTS = "similarInterestsGenerator"
 
 
 class SimilarInterestsGenerator(MatchGenerator):
     def __init__(self, min_common: int = 0):
-        # Minimum number of interests users must have in common to be eligible
-        # for a similar interests match.
         self.min_common = min_common
         super().__init__(name=GENERATOR_SIMILAR_INTERESTS)
 
     def generate(self, inp: MatchingInput) -> Iterator[Match]:
-        inp.logger.info("Bridget was here!")
-        yield from []
+        listUsers = inp.users
+        if listUsers == []:
+            matchUsers = {"Empty", "Empty"}
+            metadata = MatchMetadata(
+                generator=GENERATOR_SIMILAR_INTERESTS,
+                score=0.0,
+                commonInterests=[],
+            )
+            yield Match(users=matchUsers, metadata=metadata)
```

Comment:
You don't have to yield an empty match if there are no users. In fact, this will cause an error in the matching engine because `"Empty"` will not be a valid user ID.

In this case, you can either do:

```python
if listUsers == []:
  return iter([])
```

```python
if listUsers == []:
  return
  yield
```

More discussion on [this StackOverflow post](https://stackoverflow.com/a/6266586).